modules/apps/portal-search/portal-search-admin-web/src/main/resources/META-INF/resources/js/IndexActions.js

```diff
@@ -0,0 +1,628 @@
+						className="save-server-button"
+						disabled={disabled}
+						displayType="secondary"
+						onClick={_handleClick}
+					>
+						{Liferay.Language.get('reindex')}
+					</ClayButton>
+				)}
+			</ClayList.ItemField>
+		</ClayList.Item>
+	);
+}
+
+function IndexActions({
+	controlMenuCategoryKey,
+	elasticSearchDiskSpace = {
+		availableDiskSpace: 0,
+		currentDiskSpaceUsed: 0,
+		isLowOnDiskSpace: false,
+	},
+	indexersMap = {},
+	initialCompanyIds,
+	initialExecutionMode,
+	initialScope,
+	isConcurrentModeSupported = true,
+	virtualInstances = [],
+	indexReindexerNames = [],
+	portletNamespace,
+	redirectURL = '',
+	reindexURL = '',
+}) {
+	const [backgroundTaskMap, setBackgroundTaskMap] = useState({});
+	const [selectedCompanyIds, setSelectedCompanyIds] = useState(
+		initialCompanyIds || []
+	);
+	const [executionMode, setExecutionMode] = useState(
+		initialExecutionMode || EXECUTION_MODES.FULL.value
+	);
+	const [executionScope, setExecutionScope] = useState(
+		initialScope || SCOPES.ALL
+	);
+
+	const [state, dispatch] = useContext(ClayModalContext);
+
+	const intervalRef = useRef(null);
+
+	/*
+	 * Returns the list of virtual instances to reindex, depending on the
+	 * `executionScope` and `selectedCompanyIds`.
+	 */
+	const _getCompanyIds = () => {
+		return executionScope === SCOPES.ALL
+			? virtualInstances.map(({id}) => id)
+			: selectedCompanyIds;
+	};
+
+	/*
+	 * Executed after performing the reindex call. Continuously fetches the
+	 * view of the current URL to get status of the background task.
+	 * Calls `_handleBackgroundTaskStop` within when there are no more
+	 * background tasks running.
+	 */
+	const _handleBackgroundTaskStart = () => {
+		const fetchRedirectURL = new URL(redirectURL);
+
+		if (!intervalRef.current) {
+			intervalRef.current = setInterval(() => {
+				fetch(fetchRedirectURL, {method: 'GET'})
+					.then((response) => response.text())
+					.then((text) => {
+						const parser = new DOMParser();
+						const htmlDocument = parser.parseFromString(
+							text,
+							'text/html'
+						);
+						const backgroundTaskMapString = htmlDocument.documentElement
+							.querySelector(
+								`#${portletNamespace}classNameToBackgroundTaskMap`
+							)
+							?.innerHTML?.trim();
+
+						const newBackgroundTaskMap =
+							JSON.parse(backgroundTaskMapString) || {};
+
+						if (Object.keys(newBackgroundTaskMap).length) {
+							setBackgroundTaskMap(newBackgroundTaskMap);
+						}
+						else {
+							_handleBackgroundTaskStop();
+
+							_handleSyncIconRemove();
+
+							openToast({
+								message: Liferay.Language.get(
+									'reindexing-finished-successfully'
+								),
+								type: 'success',
+							});
+						}
+					})
+					.catch(() => {
+						_handleBackgroundTaskStop();
+
+						_handleSyncIconRemove();
+
+						openToast({
+							message: Liferay.Language.get(
+								'an-unexpected-error-occurred'
+							),
+							type: 'danger',
+						});
+					});
+			}, 1000);
```

Comment:
Let's decrease the refresh to 2000 (which is what it was before `const INTERVAL_RENDER_IN_PROGRESS = 2000;`) . 1s feels like it'd be hitting the servers too often especially since this loads a whole page.